src/Distributor.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity =0.8.11;
+
+import "@openzeppelin/access/Ownable.sol";
+import "@openzeppelin/interfaces/IERC20.sol";
+
+/// @param recipient The address to receive the share of incomingToken
+/// @param weight The share of incoming tokens in basis points
+/// @dev The sum of all weights should be BASIS_POINTS_MAX
+struct Allocation
+{
+    address recipient;
+    uint16 weight;
+}
+
+contract Distributor is Ownable
+{
+    uint16 public constant BASIS_POINTS_MAX = 10000;
+
+    IERC20 public immutable incomingToken;
+    Allocation[] public allocations;
+
+    constructor(IERC20 aIncomingToken)
+    {
+        incomingToken = aIncomingToken;
+    }
+
+    function getAllocationsLength() external view returns (uint256)
+    { 
+        return allocations.length;
+    }
+
+    function getAllocation(uint256 aIndex) external view returns (Allocation memory)
+    {
+        return allocations[aIndex];
+    }
+
+    function sumWeights(Allocation[] memory aAllocations) internal pure returns (uint256 lSum)
+    {
+        lSum = 0;
+        for(uint256 i = 0; i < aAllocations.length; ++i) {
+            lSum += aAllocations[i].weight;
+        }
+    }
```

Comment:
In vaults & generally, I propose using `rVariable` for return scoped variables:
```suggestion
    function sumWeights(Allocation[] memory aAllocations) internal pure returns (uint256 rSum)
    {
        rSum = 0;
        for(uint256 i = 0; i < aAllocations.length; ++i) {
            rSum += aAllocations[i].weight;
        }
    }
```

Reply:
fixed